shared-libs/cht-datasource/test/remote/contact.spec.ts

```diff
@@ -0,0 +1,111 @@
+  let getResourceInner: SinonStub;
+  let getResourceOuter: SinonStub;
+  let getResourcesInner: SinonStub;
+  let getResourcesOuter: SinonStub;
+
+  beforeEach(() => {
+    getResourceInner = sinon.stub();
+    getResourceOuter = sinon.stub(RemoteEnv, 'getResource').returns(getResourceInner);
+    getResourcesInner = sinon.stub();
+    getResourcesOuter = sinon.stub(RemoteEnv, 'getResources').returns(getResourcesInner);
+  });
+
+  afterEach(() => sinon.restore());
+
+  describe('v1', () => {
+    const identifier = {uuid: 'uuid'} as const;
+
+    describe('get', () => {
+      it('returns a contact by UUID', async () => {
+        const doc = { type: 'person' };
+        getResourceInner.resolves(doc);
+
+        const result = await Contact.v1.get(remoteContext)(identifier);
+
+        expect(result).to.equal(doc);
+        expect(getResourceOuter.calledOnceWithExactly(remoteContext, 'api/v1/contact')).to.be.true;
+        expect(getResourceInner.calledOnceWithExactly(identifier.uuid)).to.be.true;
+      });
+
+      it('returns null if the identified doc is not found', async () => {
+        getResourceInner.resolves(null);
+
+        const result = await Contact.v1.get(remoteContext)(identifier);
+
+        expect(result).to.be.null;
+        expect(getResourceOuter.calledOnceWithExactly(remoteContext, 'api/v1/contact')).to.be.true;
+        expect(getResourceInner.calledOnceWithExactly(identifier.uuid)).to.be.true;
+      });
+    });
+
+    describe('getWithLineage', () => {
+      it('returns a contact with lineage by UUID', async () => {
+        const doc = { type: 'person' };
+        getResourceInner.resolves(doc);
+
+        const result = await Contact.v1.getWithLineage(remoteContext)(identifier);
+
+        expect(result).to.equal(doc);
+        expect(getResourceOuter.calledOnceWithExactly(remoteContext, 'api/v1/contact')).to.be.true;
+        expect(getResourceInner.calledOnceWithExactly(identifier.uuid, { with_lineage: 'true' })).to.be.true;
+      });
+
+      it('returns null if the identified doc is not found', async () => {
+        getResourceInner.resolves(null);
+
+        const result = await Contact.v1.getWithLineage(remoteContext)(identifier);
+
+        expect(result).to.be.null;
+        expect(getResourceOuter.calledOnceWithExactly(remoteContext, 'api/v1/contact')).to.be.true;
+        expect(getResourceInner.calledOnceWithExactly(identifier.uuid, { with_lineage: 'true' })).to.be.true;
+      });
+    });
+
+    describe('getPage', () => {
+      const limit = 3;
+      const cursor = '1';
+      const contactType = 'person';
+      const freetext = 'contact';
+      const qualifier = {
+        contactType,
+        freetext
+      };
+      const queryParam = {
+        limit: limit.toString(),
+        freetext: freetext,
+        type: contactType,
+        cursor,
+      };
+
+      it('returns contacts', async () => {
+        const doc = [{ type: 'person' }, {type: 'person'}];
```

Comment:
Ok, so this calls `getIdsPage` which returns a list of ids? Not contacts? I keep getting lost in return types.